docs/solr-cloud/solr-cloud-crd.md

```diff
@@ -543,3 +543,234 @@ spec:
+  ...
+  solrSecurity:
+    authenticationType: Basic
+    probesRequireAuth: true
+```
+When `probesRequireAuth` is set to `true`, the liveness and readiness probes execute a command instead of using HTTP. 
+The operator configures a command instead of setting the `Authorization` header for the HTTP probes, as that would require a restart of all pods if the password changes. 
+With a command, we can load the username and password from a secret; Kubernetes will 
+[update the mounted secret files](https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically) when the secret changes automatically.
+
+#### Authorization
+
+The default `security.json` created by the operator during initialization is shown below; the passwords for each user are randomized for every SolrCloud you create.
+In addition to configuring the `solr.BasicAuthPlugin`, the operator initializes a set of authorization rules for the default user accounts: `admin`, `solr`, and `k8s-oper`.
+Take a moment to review these authorization rules so that you're aware of the roles and access granted to each user in your cluster.
+```
+{
+  "authentication": {
+    "blockUnknown": false,
+    "class": "solr.BasicAuthPlugin",
+    "credentials": {
+      "admin": "...",
+      "k8s-oper": "...",
+      "solr": "..."
+    },
+    "realm": "Solr Basic Auth",
+    "forwardCredentials": false
+  },
+  "authorization": {
+    "class": "solr.RuleBasedAuthorizationPlugin",
+    "user-role": {
+      "admin": [ "admin", "k8s" ],
+      "k8s-oper": [ "k8s" ],
+      "solr": [ "users", "k8s" ]
+    },
+    "permissions": [
+      {
+        "name": "k8s-probe-0",
+        "role": null,
+        "collection": null,
+        "path": "/admin/info/system"
+      },
+      {
+        "name": "k8s-status",
+        "role": "k8s",
+        "collection": null,
+        "path": "/admin/collections"
+      },
+      {
+        "name": "k8s-metrics",
+        "role": "k8s",
+        "collection": null,
+        "path": "/admin/metrics"
+      },
+      {
+        "name": "k8s-ping",
+        "role": "k8s",
+        "collection": "*",
+        "path": "/admin/ping"
+      },
+      {
+        "name": "all",
+        "role": [ "admin", "users" ]
+      },
+      {
+        "name": "read",
+        "role": [ "admin", "users" ]
+      },
+      {
+        "name": "update",
+        "role": [ "admin" ]
+      },
+      {
+        "name": "security-read",
+        "role": "admin"
+      },
+      {
+        "name": "security-edit",
+        "role": "admin"
+      }
+    ]
+  }
+}
+```
+A few aspects of the default `security.json` configuration warrant a closer look. First, the `probesRequireAuth` setting 
+(defaults to `false`) governs the value for `blockUnknown` (under `authentication`) and whether the probe endpoint(s) require authentication:
+```
+      {
+        "name": "k8s-probe-0",
+        "role": null,
+        "collection": null,
+        "path": "/admin/info/system"
```

Comment:
We probably want to add `/admin/info/health` for the future as well. (this would affect multiple places in the PR)

Reply:
We do you think we should do about the `customPodOptions` probes and auth? Currently, if the user supplies a custom  probe config with a different path and has `probesRequireAuth`, the code is not picking up the custom path. If `probesRequireAuth` is `false`, then the custom path does get added to the authz rules correctly. This is a bit trappy but I don't know if the operator should try to swap out their custom probe config with a command either? So perhaps we just doc this limitation for now?

Reply:
If we go with the punt on `probesRequireAuth` and custom probe paths approach, I think we should also error out during reconcile so it's clear we don't like their config, i.e. something along the lines of:
```
Found custom probe HTTP paths and probesRequireAuth=true, this configuration is not currently supported. Please switch to probesRequireAuth=false when using custom probe HTTP paths.
```

Reply:
I've included `/admin/info/health` as a default in the rules for now

Reply:
Re-worked this a bit and improved the docs in this area. Added a check for how custom probe paths are handled to one of the existing unit tests